src/logic/hooks/useEstimateSafeCreationGas.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { userAccountSelector } from '../wallets/store/selectors'
+import { estimateGasForDeployingSafe } from 'src/logic/contracts/safeContracts'
+import { fromTokenUnit } from 'src/logic/tokens/utils/humanReadableValue'
+import { formatAmount } from 'src/logic/tokens/utils/formatAmount'
+import { getNetworkInfo } from 'src/config'
+import { calculateGasPrice } from 'src/logic/wallets/ethTransactions'
+import { MINIMUM_TRANSACTION_GAS } from 'src/logic/safe/transactions/gas'
+
+type EstimateSafeCreationGasProps = {
+  addresses: string[]
+  numOwners: number
+  safeCreationSalt: number
+}
+
+type SafeCreationEstimationResult = {
+  gasEstimation: number // Amount of gas needed for execute or approve the transaction
+  gasCostFormatted: string // Cost of gas in format '< | > 100'
+  gasLimit: number // Minimum gas requited to execute the Tx
+}
+
+const { nativeCoin } = getNetworkInfo()
+
+export const useEstimateSafeCreationGas = ({
+  addresses,
+  numOwners,
+  safeCreationSalt,
+}: EstimateSafeCreationGasProps): SafeCreationEstimationResult => {
+  const [gasEstimation, setGasEstimation] = useState<SafeCreationEstimationResult>({
+    gasEstimation: 0,
+    gasCostFormatted: '< 0.001',
+    gasLimit: 0,
+  })
+  const userAccount = useSelector(userAccountSelector)
+
+  useEffect(() => {
+    const estimateGas = async () => {
+      if (!addresses.length || !numOwners || !userAccount) {
+        return
+      }
+
+      const gasEstimation = await estimateGasForDeployingSafe(addresses, numOwners, userAccount, safeCreationSalt)
+      const gasPrice = await calculateGasPrice()
+      const estimatedGasCosts = gasEstimation * parseInt(gasPrice, 10)
+      const gasCost = fromTokenUnit(estimatedGasCosts, nativeCoin.decimals)
+      const gasCostFormatted = formatAmount(gasCost)
+      const gasLimit = gasEstimation * 2 + MINIMUM_TRANSACTION_GAS
```

Comment:
this `* 2` it's not required. `estimateGasForDeployingSafe` it's already doing the multiplication. 
I tested it creating a safe with 6 owners and the estimation was correct.
https://rinkeby.etherscan.io/tx/0xb8d60d184d8244e1d8906222d4d83b2209677ac549e8f10bb8b2cad49d8848a1

@dasanra Am I correct?